pkg/cluster/cluster.go

```diff
@@ -211,11 +211,10 @@ func (c *Cluster) run(stopC <-chan struct{}, wg *sync.WaitGroup) {
 		if needDeleteCluster {
 			c.logger.Infof("deleting cluster")
 			c.delete()
+			go c.reportFailedStatus()
```

Comment:
if reconcile fails => needDeleteCluster = true

the tpr might already be deleted => See reportFailedStatus() impl. It will return if TPR deleted